pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -31,16 +32,43 @@ type StepResolver interface {
 }
 
 type OperatorStepResolver struct {
-	subLister              v1alpha1listers.SubscriptionLister
-	csvLister              v1alpha1listers.ClusterServiceVersionLister
-	client                 versioned.Interface
-	globalCatalogNamespace string
-	resolver               *Resolver
-	log                    logrus.FieldLogger
+	subLister                 v1alpha1listers.SubscriptionLister
+	csvLister                 v1alpha1listers.ClusterServiceVersionLister
+	client                    versioned.Interface
+	globalCatalogNamespace    string
+	resolver                  *Resolver
+	log                       logrus.FieldLogger
+	globalCatalogSourceToggle GlobalNamespaceToggle
 }
 
 var _ StepResolver = &OperatorStepResolver{}
 
+type GlobalNamespaceToggle interface {
+	ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error)
+}
+
+func NewNamespaceAnnotationGlobalCatalogToggle(opClient operatorclient.ClientInterface) *NamespaceAnnotationGlobalCatalogToggle {
+	return &NamespaceAnnotationGlobalCatalogToggle{
+		opClient: opClient,
+	}
+}
+
+type NamespaceAnnotationGlobalCatalogToggle struct {
+	opClient operatorclient.ClientInterface
+}
+
+func (n *NamespaceAnnotationGlobalCatalogToggle) ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error) {
+	fmt.Println("[PING] Getting global catalog source toggle")
+	ns, err := n.opClient.KubernetesInterface().CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+	fmt.Printf("Got namespace %s and error %s", ns, err)
+	if err != nil {
+		return false, err
+	}
+	_, ok := ns.Annotations["olm.disableGlobalCatalogs"]
```

Comment:
Maybe we should consider the value of the annotation as well? For example, if a user sets it to `true`, then we would disable resolution from the global catalogs. But if the same user were to later decide that they did want resolution from the global catalogs as well, they could set the annotation to anything else (?) and then can opt-in to global resolution again? 

But I'm not opposed to the existing implementation where we would just say "put this annotation key on the namespace, regardless of the value it will work as expected". It just seems like maybe if they set the value to false they may expect the behavior to revert, which it would not.